src/App.tsx

```diff
@@ -1,24 +1,266 @@
+              ...todo,
+              completed: !todo.completed,
+            };
+
+            updateTodoStatus(todoId, updatedTodo.completed);
+
+            return updatedTodo;
+          }
+
+          return todo;
+        });
+
+        setTodos(updatedTodos);
+      } catch {
+        setError('Unable to update a todo');
+      }
+    },
+    [todos],
+  );
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setIsDisabled(true);
+      try {
+        const arrTodos = await fetchTodos(USER_ID.toString());
+
+        setTodos(arrTodos);
+        setIsDisabled(false);
+      } catch {
+        setError('Unable to fetch todos');
+      }
+    };
+
+    if (USER_ID) {
+      fetchData();
+    }
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    switch (filter) {
+      case Filter.ACTIVE:
+        return todos.filter(todo => !todo.completed);
+
+      case Filter.COMPLETED:
+        return todos.filter(todo => todo.completed);
+
+      case Filter.ALL:
+        return todos;
+
+      default:
+        return todos;
+    }
+  }, [filter, todos]);
+
+  const handleClearCompleted = useCallback(() => {
+    const completedIds = todos
+      .filter(todo => todo.completed)
+      .map(todo => todo.id);
+
+    setIsDisabled(true);
+
+    removeTodos(completedIds)
+      .then(() => {
+        setTodos(prev => prev.filter(todo => !todo.completed));
+        setIsDisabled(false);
+      })
+      .catch(() => {
+        setError('Unable to delete todos');
+      });
+  }, [todos]);
+
+  const updateTodoTitle = async (id: number, newTitle: string) => {
+    try {
+      const updatedTodos = todos.map((todo) => {
+        if (todo.id === id) {
+          return {
+            ...todo,
+            title: newTitle,
+          };
+        }
+
+        return todo;
+      });
+
+      setTodos(updatedTodos);
+      await updateTodo(id, newTitle);
+    } catch {
+      setError('Unable to update a todo');
+    }
+  };
+
+  const handleToggleAll = useCallback(() => {
+    const allCompleted = todos.every((todo) => todo.completed);
+
+    const updatedTodos = todos.map((todo) => ({
+      ...todo,
+      completed: !allCompleted,
+    }));
+
+    updateTodoStatuses(updatedTodos)
+      .then(() => {
+        setTodos(updatedTodos);
+      })
+      .catch(() => {
+        setError('Unable to complete todos');
+      });
+  }, [todos]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+
+        <Header
```

Comment:
```suggestion
      <div className="todoapp__content">
        <Header
```
Don't add empty lines between parent and child elements